Use template specializations for Markable Traits

cdumez opened the pull request:
#### 276b1deb070bc5b2a055dd8b301908af8bf62e08
<pre>
Use template specializations for Markable Traits
<a href="https://bugs.webkit.org/show_bug.cgi?id=295592">https://bugs.webkit.org/show_bug.cgi?id=295592</a>

Reviewed by Darin Adler and Dan Glastonbury.

Use template specialization for Markable Traits. This avoids having to
specify the second template parameter for Markable in more cases.

* Source/JavaScriptCore/heap/Heap.h:
* Source/JavaScriptCore/runtime/FunctionExecutable.h:
* Source/WTF/wtf/ApproximateTime.h:
(WTF::ApproximateTime::MarkableTraits::isEmptyValue): Deleted.
(WTF::ApproximateTime::MarkableTraits::emptyValue): Deleted.
* Source/WTF/wtf/ContinuousApproximateTime.h:
(WTF::ContinuousApproximateTime::MarkableTraits::isEmptyValue): Deleted.
(WTF::ContinuousApproximateTime::MarkableTraits::emptyValue): Deleted.
* Source/WTF/wtf/ContinuousTime.h:
(WTF::ContinuousTime::MarkableTraits::isEmptyValue): Deleted.
(WTF::ContinuousTime::MarkableTraits::emptyValue): Deleted.
* Source/WTF/wtf/Forward.h:
* Source/WTF/wtf/Markable.h:
(WTF::MarkableTraits&lt;T&gt;::isEmptyValue):
(WTF::MarkableTraits&lt;T&gt;::emptyValue):
(WTF::FloatMarkableTraits::isEmptyValue): Deleted.
(WTF::FloatMarkableTraits::emptyValue): Deleted.
* Source/WTF/wtf/MonotonicTime.h:
(WTF::MonotonicTime::MarkableTraits::isEmptyValue): Deleted.
(WTF::MonotonicTime::MarkableTraits::emptyValue): Deleted.
* Source/WTF/wtf/ObjectIdentifier.h:
(WTF::ObjectIdentifierGeneric::MarkableTraits::isEmptyValue): Deleted.
(WTF::ObjectIdentifierGeneric::MarkableTraits::emptyValue): Deleted.
* Source/WTF/wtf/Seconds.h:
(WTF::Seconds::MarkableTraits::isEmptyValue): Deleted.
(WTF::Seconds::MarkableTraits::emptyValue): Deleted.
* Source/WTF/wtf/UUID.h:
(WTF::MarkableTraits&lt;UUID&gt;::isEmptyValue):
(WTF::MarkableTraits&lt;UUID&gt;::emptyValue):
(WTF::UUID::MarkableTraits::isEmptyValue): Deleted.
(WTF::UUID::MarkableTraits::emptyValue): Deleted.
* Source/WTF/wtf/WallTime.h:
(WTF::WallTime::MarkableTraits::isEmptyValue): Deleted.
(WTF::WallTime::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/animation/AnimationEffectTiming.h:
* Source/WebCore/animation/AnimationEventBase.h:
* Source/WebCore/animation/AnimationTimelinesController.h:
* Source/WebCore/animation/ComputedEffectTiming.h:
* Source/WebCore/animation/FillMode.h:
* Source/WebCore/animation/OptionalEffectTiming.h:
* Source/WebCore/animation/PlaybackDirection.h:
* Source/WebCore/animation/WebAnimation.h:
* Source/WebCore/animation/WebAnimationTypes.h:
(WebCore::WebAnimationsMarkableDoubleTraits::isEmptyValue): Deleted.
(WebCore::WebAnimationsMarkableDoubleTraits::emptyValue): Deleted.
* Source/WebCore/css/typedom/numeric/CSSNumericType.h:
* Source/WebCore/css/values/color/CSSColor.cpp:
(WebCore::CSS::Color::MarkableTraits::isEmptyValue): Deleted.
(WebCore::CSS::Color::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/css/values/color/CSSColor.h:
(WTF::MarkableTraits&lt;WebCore::CSS::Color&gt;::isEmptyValue):
(WTF::MarkableTraits&lt;WebCore::CSS::Color&gt;::emptyValue):
* Source/WebCore/css/values/primitives/CSSPrimitiveNumeric.h:
* Source/WebCore/css/values/primitives/CSSPrimitiveNumericOrKeyword.h:
(WebCore::CSS::PrimitiveNumericOrKeyword::MarkableTraits::isEmptyValue): Deleted.
(WebCore::CSS::PrimitiveNumericOrKeyword::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/css/values/primitives/CSSPrimitiveNumericTypes.h:
(WebCore::CSS::NumberOrPercentage::MarkableTraits::isEmptyValue): Deleted.
(WebCore::CSS::NumberOrPercentage::MarkableTraits::emptyValue): Deleted.
(WebCore::CSS::NumberOrPercentageResolvedToNumber::MarkableTraits::isEmptyValue): Deleted.
(WebCore::CSS::NumberOrPercentageResolvedToNumber::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/dom/ElementRareData.cpp:
* Source/WebCore/dom/ElementRareData.h:
(WebCore::LayoutUnitMarkableTraits::isEmptyValue): Deleted.
(WebCore::LayoutUnitMarkableTraits::emptyValue): Deleted.
* Source/WebCore/dom/NodeWithIndex.h:
* Source/WebCore/fileapi/URLKeepingBlobAlive.h:
* Source/WebCore/html/HTMLAnchorElement.h:
* Source/WebCore/html/LinkRelAttribute.h:
* Source/WebCore/loader/ResourceLoaderOptions.h:
* Source/WebCore/page/SecurityOriginData.h:
(WTF::MarkableTraits&lt;WebCore::SecurityOriginData&gt;::isEmptyValue):
(WTF::MarkableTraits&lt;WebCore::SecurityOriginData&gt;::emptyValue):
(WebCore::SecurityOriginDataMarkableTraits::isEmptyValue): Deleted.
(WebCore::SecurityOriginDataMarkableTraits::emptyValue): Deleted.
* Source/WebCore/page/csp/ContentSecurityPolicyResponseHeaders.h:
(WTF::MarkableTraits&lt;WebCore::ContentSecurityPolicyResponseHeaders&gt;::isEmptyValue):
(WTF::MarkableTraits&lt;WebCore::ContentSecurityPolicyResponseHeaders&gt;::emptyValue):
(WebCore::ContentSecurityPolicyResponseHeaders::MarkableTraits::isEmptyValue): Deleted.
(WebCore::ContentSecurityPolicyResponseHeaders::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/platform/LayoutUnit.h:
(WTF::MarkableTraits&lt;WebCore::LayoutUnit&gt;::isEmptyValue):
(WTF::MarkableTraits&lt;WebCore::LayoutUnit&gt;::emptyValue):
* Source/WebCore/platform/Length.h:
(WTF::MarkableTraits&lt;WebCore::Length&gt;::isEmptyValue):
(WTF::MarkableTraits&lt;WebCore::Length&gt;::emptyValue):
(WebCore::Length::MarkableTraits::isEmptyValue): Deleted.
(WebCore::Length::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/platform/ProcessQualified.h:
(WTF::MarkableTraits&lt;WebCore::ProcessQualified&lt;T&gt;&gt;::isEmptyValue):
(WTF::MarkableTraits&lt;WebCore::ProcessQualified&lt;T&gt;&gt;::emptyValue):
(WebCore::ProcessQualified::MarkableTraits::isEmptyValue): Deleted.
(WebCore::ProcessQualified::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/platform/ScriptExecutionContextIdentifier.h:
(WTF::MarkableTraits&lt;WebCore::ProcessQualified&lt;UUID&gt;&gt;::isEmptyValue):
(WTF::MarkableTraits&lt;WebCore::ProcessQualified&lt;UUID&gt;&gt;::emptyValue):
(WebCore::ProcessQualified&lt;WTF::UUID&gt;::MarkableTraits::isEmptyValue): Deleted.
(WebCore::ProcessQualified&lt;WTF::UUID&gt;::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/platform/SharedStringHash.h:
* Source/WebCore/platform/animation/Animation.h:
(WebCore::Animation::duration const):
(WebCore::Animation::setDuration):
(WebCore::Animation::fillDuration):
(WebCore::Animation::initialDuration):
* Source/WebCore/platform/graphics/FloatPoint.h:
(WTF::MarkableTraits&lt;WebCore::FloatPoint&gt;::isEmptyValue):
(WTF::MarkableTraits&lt;WebCore::FloatPoint&gt;::emptyValue):
(WebCore::FloatPoint::MarkableTraits::isEmptyValue): Deleted.
(WebCore::FloatPoint::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/platform/graphics/FloatRect.h:
(WTF::MarkableTraits&lt;WebCore::FloatRect&gt;::isEmptyValue):
(WTF::MarkableTraits&lt;WebCore::FloatRect&gt;::emptyValue):
(WebCore::FloatRect::MarkableTraits::isEmptyValue): Deleted.
(WebCore::FloatRect::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/platform/graphics/FloatSize.h:
(WTF::MarkableTraits&lt;WebCore::FloatSize&gt;::isEmptyValue):
(WTF::MarkableTraits&lt;WebCore::FloatSize&gt;::emptyValue):
(WebCore::FloatSize::MarkableTraits::isEmptyValue): Deleted.
(WebCore::FloatSize::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/platform/graphics/FontMetrics.h:
(WebCore::FontMetrics::xHeight const):
(WebCore::FontMetrics::capHeight const):
(WebCore::FontMetrics::zeroWidth const):
(WebCore::FontMetrics::ideogramWidth const):
(WebCore::FontMetrics::underlinePosition const):
(WebCore::FontMetrics::underlineThickness const):
(WebCore::FontMetrics::MarkableTraits::isEmptyValue): Deleted.
(WebCore::FontMetrics::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/platform/graphics/FontSizeAdjust.h:
* Source/WebCore/platform/network/CacheValidation.h:
* Source/WebCore/platform/network/ParsedContentRange.h:
(WTF::MarkableTraits&lt;WebCore::ParsedContentRange&gt;::isEmptyValue):
(WTF::MarkableTraits&lt;WebCore::ParsedContentRange&gt;::emptyValue):
(WebCore::ParsedContentRange::MarkableTraits::isEmptyValue): Deleted.
(WebCore::ParsedContentRange::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/platform/network/ResourceResponseBase.h:
* Source/WebCore/platform/text/TextFlags.h:
(WTF::MarkableTraits&lt;WebCore::FontVariantAlternatesValues&gt;::isEmptyValue):
(WTF::MarkableTraits&lt;WebCore::FontVariantAlternatesValues&gt;::emptyValue):
(WebCore::FontVariantAlternatesValues::MarkableTraits::isEmptyValue): Deleted.
(WebCore::FontVariantAlternatesValues::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/rendering/RenderText.h:
* Source/WebCore/rendering/style/ScrollbarColor.h:
(WTF::MarkableTraits&lt;WebCore::ScrollbarColor&gt;::isEmptyValue):
(WTF::MarkableTraits&lt;WebCore::ScrollbarColor&gt;::emptyValue):
(WebCore::ScrollbarColor::MarkableTraits::isEmptyValue): Deleted.
(WebCore::ScrollbarColor::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/rendering/svg/RenderSVGContainer.h:
* Source/WebCore/rendering/svg/RenderSVGRoot.h:
* Source/WebCore/rendering/svg/RenderSVGShape.h:
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGContainer.h:
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGRoot.h:
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGShape.h:
* Source/WebCore/style/StyleExtractorConverter.h:
(WebCore::Style::ExtractorConverter::convertAnimationDuration):
* Source/WebCore/style/StyleExtractorSerializer.h:
(WebCore::Style::ExtractorSerializer::serializeAnimationDuration):
* Source/WebCore/style/values/color/StyleColor.cpp:
(WebCore::Style::Color::MarkableTraits::isEmptyValue): Deleted.
(WebCore::Style::Color::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/style/values/color/StyleColor.h:
(WTF::MarkableTraits&lt;WebCore::Style::Color&gt;::isEmptyValue):
(WTF::MarkableTraits&lt;WebCore::Style::Color&gt;::emptyValue):
* Source/WebCore/style/values/primitives/StylePrimitiveNumeric.h:
* Source/WebCore/style/values/primitives/StylePrimitiveNumericTypes.h:
(WebCore::Style::NumberOrPercentage::MarkableTraits::isEmptyValue): Deleted.
(WebCore::Style::NumberOrPercentage::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/workers/service/background-fetch/BackgroundFetchManager.cpp:
(WebCore::BackgroundFetchManager::fetch):
* Source/WebCore/workers/service/background-fetch/BackgroundFetchRequest.h:
* Source/WebGPU/WGSL/Types.h:
* Source/WebKit/Shared/WebHitTestResultData.h:
(WTF::MarkableTraits&lt;WebKit::WebHitTestResultPlatformData::DetectedDataActionContext&gt;::isEmptyValue):
(WTF::MarkableTraits&lt;WebKit::WebHitTestResultPlatformData::DetectedDataActionContext&gt;::emptyValue):
(WebKit::WebHitTestResultPlatformData::DetectedDataActionContext::MarkableTraits::isEmptyValue): Deleted.
(WebKit::WebHitTestResultPlatformData::DetectedDataActionContext::MarkableTraits::emptyValue): Deleted.
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeNode.h:
* Tools/TestWebKitAPI/Tests/WebCore/FloatPointTests.cpp:
(TestWebKitAPI::TEST(FloatPoint, Markable)):
* Tools/TestWebKitAPI/Tests/WebCore/FloatRectTests.cpp:
(TestWebKitAPI::TEST(FloatRect, Markable)):
* Tools/TestWebKitAPI/Tests/WebCore/FloatSizeTests.cpp:
(TestWebKitAPI::TEST(FloatSize, Markable)):

Canonical link: <a href="https://commits.webkit.org/297154@main">https://commits.webkit.org/297154@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/845af279b2303033848bfe6b781450ec4285d134

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/110696 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/30355 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/20791 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/116723 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/60963 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/112659 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/31034 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/38944 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/84173 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/113644 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/24792 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/99682 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/64614 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/24157 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/17820 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/60517 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/103188 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/94182 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/17879 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/119513 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/109250 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/37736 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/28039 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/93139 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/38110 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/95950 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/92963 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23692 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/37990 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/15737 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/33714 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/37632 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/43104 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/133525 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/37294 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/36068 "Found 1 new JSC stress test failure: wasm.yaml/wasm/stress/jitless-wasm-to-js-should-follow-to-wasm-call-frame-convention-2.js.default (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/40633 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/39002 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

djg reviewed (approved):
:shipit:

darinadler commented on Source/WTF/wtf/Markable.h:
```diff
@@ -41,6 +41,8 @@
 
 namespace WTF {
 
+template<typename T> struct MarkableTraits { };
```
```suggestion
template<typename> struct MarkableTraits { };
```

cdumez commented on Source/WTF/wtf/Markable.h:
```diff
@@ -62,7 +64,11 @@ struct EnumMarkableTraits {
 
-template<typename IntegralType, IntegralType constant = 0>
+template<typename T>
+requires(std::is_scoped_enum_v<T>)
```
`std:: is_scoped_enum_v` is not supported by the windows port. I'll work on an alternative.

cdumez committed:
Use template specializations for Markable Traits
https://bugs.webkit.org/show_bug.cgi?id=295592

Reviewed by Darin Adler and Dan Glastonbury.

Use template specialization for Markable Traits. This avoids having to
specify the second template parameter for Markable in more cases.

* Source/JavaScriptCore/heap/Heap.h:
* Source/JavaScriptCore/runtime/FunctionExecutable.h:
* Source/WTF/wtf/ApproximateTime.h:
(WTF::ApproximateTime::MarkableTraits::isEmptyValue): Deleted.
(WTF::ApproximateTime::MarkableTraits::emptyValue): Deleted.
* Source/WTF/wtf/ContinuousApproximateTime.h:
(WTF::ContinuousApproximateTime::MarkableTraits::isEmptyValue): Deleted.
(WTF::ContinuousApproximateTime::MarkableTraits::emptyValue): Deleted.
* Source/WTF/wtf/ContinuousTime.h:
(WTF::ContinuousTime::MarkableTraits::isEmptyValue): Deleted.
(WTF::ContinuousTime::MarkableTraits::emptyValue): Deleted.
* Source/WTF/wtf/Forward.h:
* Source/WTF/wtf/Markable.h:
(WTF::MarkableTraits<T>::isEmptyValue):
(WTF::MarkableTraits<T>::emptyValue):
(WTF::FloatMarkableTraits::isEmptyValue): Deleted.
(WTF::FloatMarkableTraits::emptyValue): Deleted.
* Source/WTF/wtf/MonotonicTime.h:
(WTF::MonotonicTime::MarkableTraits::isEmptyValue): Deleted.
(WTF::MonotonicTime::MarkableTraits::emptyValue): Deleted.
* Source/WTF/wtf/ObjectIdentifier.h:
(WTF::ObjectIdentifierGeneric::MarkableTraits::isEmptyValue): Deleted.
(WTF::ObjectIdentifierGeneric::MarkableTraits::emptyValue): Deleted.
* Source/WTF/wtf/Seconds.h:
(WTF::Seconds::MarkableTraits::isEmptyValue): Deleted.
(WTF::Seconds::MarkableTraits::emptyValue): Deleted.
* Source/WTF/wtf/UUID.h:
(WTF::MarkableTraits<UUID>::isEmptyValue):
(WTF::MarkableTraits<UUID>::emptyValue):
(WTF::UUID::MarkableTraits::isEmptyValue): Deleted.
(WTF::UUID::MarkableTraits::emptyValue): Deleted.
* Source/WTF/wtf/WallTime.h:
(WTF::WallTime::MarkableTraits::isEmptyValue): Deleted.
(WTF::WallTime::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/animation/AnimationEffectTiming.h:
* Source/WebCore/animation/AnimationEventBase.h:
* Source/WebCore/animation/AnimationTimelinesController.h:
* Source/WebCore/animation/ComputedEffectTiming.h:
* Source/WebCore/animation/FillMode.h:
* Source/WebCore/animation/OptionalEffectTiming.h:
* Source/WebCore/animation/PlaybackDirection.h:
* Source/WebCore/animation/WebAnimation.h:
* Source/WebCore/animation/WebAnimationTypes.h:
(WebCore::WebAnimationsMarkableDoubleTraits::isEmptyValue): Deleted.
(WebCore::WebAnimationsMarkableDoubleTraits::emptyValue): Deleted.
* Source/WebCore/css/typedom/numeric/CSSNumericType.h:
* Source/WebCore/css/values/color/CSSColor.cpp:
(WebCore::CSS::Color::MarkableTraits::isEmptyValue): Deleted.
(WebCore::CSS::Color::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/css/values/color/CSSColor.h:
(WTF::MarkableTraits<WebCore::CSS::Color>::isEmptyValue):
(WTF::MarkableTraits<WebCore::CSS::Color>::emptyValue):
* Source/WebCore/css/values/primitives/CSSPrimitiveNumeric.h:
* Source/WebCore/css/values/primitives/CSSPrimitiveNumericOrKeyword.h:
(WebCore::CSS::PrimitiveNumericOrKeyword::MarkableTraits::isEmptyValue): Deleted.
(WebCore::CSS::PrimitiveNumericOrKeyword::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/css/values/primitives/CSSPrimitiveNumericTypes.h:
(WebCore::CSS::NumberOrPercentage::MarkableTraits::isEmptyValue): Deleted.
(WebCore::CSS::NumberOrPercentage::MarkableTraits::emptyValue): Deleted.
(WebCore::CSS::NumberOrPercentageResolvedToNumber::MarkableTraits::isEmptyValue): Deleted.
(WebCore::CSS::NumberOrPercentageResolvedToNumber::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/dom/ElementRareData.cpp:
* Source/WebCore/dom/ElementRareData.h:
(WebCore::LayoutUnitMarkableTraits::isEmptyValue): Deleted.
(WebCore::LayoutUnitMarkableTraits::emptyValue): Deleted.
* Source/WebCore/dom/NodeWithIndex.h:
* Source/WebCore/fileapi/URLKeepingBlobAlive.h:
* Source/WebCore/html/HTMLAnchorElement.h:
* Source/WebCore/html/LinkRelAttribute.h:
* Source/WebCore/loader/ResourceLoaderOptions.h:
* Source/WebCore/page/SecurityOriginData.h:
(WTF::MarkableTraits<WebCore::SecurityOriginData>::isEmptyValue):
(WTF::MarkableTraits<WebCore::SecurityOriginData>::emptyValue):
(WebCore::SecurityOriginDataMarkableTraits::isEmptyValue): Deleted.
(WebCore::SecurityOriginDataMarkableTraits::emptyValue): Deleted.
* Source/WebCore/page/csp/ContentSecurityPolicyResponseHeaders.h:
(WTF::MarkableTraits<WebCore::ContentSecurityPolicyResponseHeaders>::isEmptyValue):
(WTF::MarkableTraits<WebCore::ContentSecurityPolicyResponseHeaders>::emptyValue):
(WebCore::ContentSecurityPolicyResponseHeaders::MarkableTraits::isEmptyValue): Deleted.
(WebCore::ContentSecurityPolicyResponseHeaders::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/platform/LayoutUnit.h:
(WTF::MarkableTraits<WebCore::LayoutUnit>::isEmptyValue):
(WTF::MarkableTraits<WebCore::LayoutUnit>::emptyValue):
* Source/WebCore/platform/Length.h:
(WTF::MarkableTraits<WebCore::Length>::isEmptyValue):
(WTF::MarkableTraits<WebCore::Length>::emptyValue):
(WebCore::Length::MarkableTraits::isEmptyValue): Deleted.
(WebCore::Length::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/platform/ProcessQualified.h:
(WTF::MarkableTraits<WebCore::ProcessQualified<T>>::isEmptyValue):
(WTF::MarkableTraits<WebCore::ProcessQualified<T>>::emptyValue):
(WebCore::ProcessQualified::MarkableTraits::isEmptyValue): Deleted.
(WebCore::ProcessQualified::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/platform/ScriptExecutionContextIdentifier.h:
(WTF::MarkableTraits<WebCore::ProcessQualified<UUID>>::isEmptyValue):
(WTF::MarkableTraits<WebCore::ProcessQualified<UUID>>::emptyValue):
(WebCore::ProcessQualified<WTF::UUID>::MarkableTraits::isEmptyValue): Deleted.
(WebCore::ProcessQualified<WTF::UUID>::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/platform/SharedStringHash.h:
* Source/WebCore/platform/animation/Animation.h:
(WebCore::Animation::duration const):
(WebCore::Animation::setDuration):
(WebCore::Animation::fillDuration):
(WebCore::Animation::initialDuration):
* Source/WebCore/platform/graphics/FloatPoint.h:
(WTF::MarkableTraits<WebCore::FloatPoint>::isEmptyValue):
(WTF::MarkableTraits<WebCore::FloatPoint>::emptyValue):
(WebCore::FloatPoint::MarkableTraits::isEmptyValue): Deleted.
(WebCore::FloatPoint::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/platform/graphics/FloatRect.h:
(WTF::MarkableTraits<WebCore::FloatRect>::isEmptyValue):
(WTF::MarkableTraits<WebCore::FloatRect>::emptyValue):
(WebCore::FloatRect::MarkableTraits::isEmptyValue): Deleted.
(WebCore::FloatRect::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/platform/graphics/FloatSize.h:
(WTF::MarkableTraits<WebCore::FloatSize>::isEmptyValue):
(WTF::MarkableTraits<WebCore::FloatSize>::emptyValue):
(WebCore::FloatSize::MarkableTraits::isEmptyValue): Deleted.
(WebCore::FloatSize::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/platform/graphics/FontMetrics.h:
(WebCore::FontMetrics::xHeight const):
(WebCore::FontMetrics::capHeight const):
(WebCore::FontMetrics::zeroWidth const):
(WebCore::FontMetrics::ideogramWidth const):
(WebCore::FontMetrics::underlinePosition const):
(WebCore::FontMetrics::underlineThickness const):
(WebCore::FontMetrics::MarkableTraits::isEmptyValue): Deleted.
(WebCore::FontMetrics::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/platform/graphics/FontSizeAdjust.h:
* Source/WebCore/platform/network/CacheValidation.h:
* Source/WebCore/platform/network/ParsedContentRange.h:
(WTF::MarkableTraits<WebCore::ParsedContentRange>::isEmptyValue):
(WTF::MarkableTraits<WebCore::ParsedContentRange>::emptyValue):
(WebCore::ParsedContentRange::MarkableTraits::isEmptyValue): Deleted.
(WebCore::ParsedContentRange::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/platform/network/ResourceResponseBase.h:
* Source/WebCore/platform/text/TextFlags.h:
(WTF::MarkableTraits<WebCore::FontVariantAlternatesValues>::isEmptyValue):
(WTF::MarkableTraits<WebCore::FontVariantAlternatesValues>::emptyValue):
(WebCore::FontVariantAlternatesValues::MarkableTraits::isEmptyValue): Deleted.
(WebCore::FontVariantAlternatesValues::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/rendering/RenderText.h:
* Source/WebCore/rendering/style/ScrollbarColor.h:
(WTF::MarkableTraits<WebCore::ScrollbarColor>::isEmptyValue):
(WTF::MarkableTraits<WebCore::ScrollbarColor>::emptyValue):
(WebCore::ScrollbarColor::MarkableTraits::isEmptyValue): Deleted.
(WebCore::ScrollbarColor::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/rendering/svg/RenderSVGContainer.h:
* Source/WebCore/rendering/svg/RenderSVGRoot.h:
* Source/WebCore/rendering/svg/RenderSVGShape.h:
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGContainer.h:
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGRoot.h:
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGShape.h:
* Source/WebCore/style/StyleExtractorConverter.h:
(WebCore::Style::ExtractorConverter::convertAnimationDuration):
* Source/WebCore/style/StyleExtractorSerializer.h:
(WebCore::Style::ExtractorSerializer::serializeAnimationDuration):
* Source/WebCore/style/values/color/StyleColor.cpp:
(WebCore::Style::Color::MarkableTraits::isEmptyValue): Deleted.
(WebCore::Style::Color::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/style/values/color/StyleColor.h:
(WTF::MarkableTraits<WebCore::Style::Color>::isEmptyValue):
(WTF::MarkableTraits<WebCore::Style::Color>::emptyValue):
* Source/WebCore/style/values/primitives/StylePrimitiveNumeric.h:
* Source/WebCore/style/values/primitives/StylePrimitiveNumericTypes.h:
(WebCore::Style::NumberOrPercentage::MarkableTraits::isEmptyValue): Deleted.
(WebCore::Style::NumberOrPercentage::MarkableTraits::emptyValue): Deleted.
* Source/WebCore/workers/service/background-fetch/BackgroundFetchManager.cpp:
(WebCore::BackgroundFetchManager::fetch):
* Source/WebCore/workers/service/background-fetch/BackgroundFetchRequest.h:
* Source/WebGPU/WGSL/Types.h:
* Source/WebKit/Shared/WebHitTestResultData.h:
(WTF::MarkableTraits<WebKit::WebHitTestResultPlatformData::DetectedDataActionContext>::isEmptyValue):
(WTF::MarkableTraits<WebKit::WebHitTestResultPlatformData::DetectedDataActionContext>::emptyValue):
(WebKit::WebHitTestResultPlatformData::DetectedDataActionContext::MarkableTraits::isEmptyValue): Deleted.
(WebKit::WebHitTestResultPlatformData::DetectedDataActionContext::MarkableTraits::emptyValue): Deleted.
* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeNode.h:
* Tools/TestWebKitAPI/Tests/WebCore/FloatPointTests.cpp:
(TestWebKitAPI::TEST(FloatPoint, Markable)):
* Tools/TestWebKitAPI/Tests/WebCore/FloatRectTests.cpp:
(TestWebKitAPI::TEST(FloatRect, Markable)):
* Tools/TestWebKitAPI/Tests/WebCore/FloatSizeTests.cpp:
(TestWebKitAPI::TEST(FloatSize, Markable)):

Canonical link: https://commits.webkit.org/297154@main
